frame/composable-traits/src/math.rs

```diff
@@ -13,32 +13,48 @@ pub fn safe_multiply_by_rational(a: u128, b: u128, c: u128) -> Result<u128, Arit
 /// Example, you sum to negative numbers, can get underflow, so need to check on each add; but if
 /// you have positive number only, you cannot have underflow. Same for other constrains, like non
 /// zero divisor.
-pub trait SafeArithmetic: Sized {
+
+pub trait SafeAdd: Sized {
 	fn safe_add(&self, rhs: &Self) -> Result<Self, ArithmeticError>;
+}
+
+pub trait SafeDiv: Sized {
 	fn safe_div(&self, rhs: &Self) -> Result<Self, ArithmeticError>;
+}
+
+pub trait SafeMul: Sized {
 	fn safe_mul(&self, rhs: &Self) -> Result<Self, ArithmeticError>;
+}
+
+pub trait SafeSub: Sized {
 	fn safe_sub(&self, rhs: &Self) -> Result<Self, ArithmeticError>;
 }
 
-impl<T: CheckedAdd + CheckedMul + CheckedDiv + CheckedSub + Zero> SafeArithmetic for T {
```

Comment:
Can we still have `SafeArithmetic`? But instead of asking for `CheckedAdd+CheckedSub..`, relying on `SafeAdd+SafeSub...`?

Reply:
@ysfkel can look into MathBalance definition - now it is hack, but on nightly they have type binding to have type A = C + B + D;